File: frontend/cars-web-app/src/interfaces/Car.ts
export enum FuelType{
    Petrol = 'Petrol',
    Hybrid = 'Hybrid',
    Diesel = 'Diesel',
    LPG = 'LPG',
};

export const FuelTypeMap: Record<number,FuelType>={
    0:FuelType.Petrol,
    1:FuelType.Hybrid,
    2:FuelType.Diesel,
    3:FuelType.LPG
};

export interface Car{
    id:string,
    brand:string,
    carFuelConsumption: number
    doorsNumber: number
    engineCapacity: number
    fuelType: FuelType
    luggageCapacity: number
    model: string
    productionDate: Date
}

File: CarsWebApplication/Cars.Domain/Car.cs
using System.ComponentModel.DataAnnotations;

namespace Cars.Domain;

public enum FuelType { Petrol, Hybrid, Diesel, LPG }
public enum BodyType { Hatchback, Sedan, Kombi, SUV, Roadster }

public class Car
{
    public Guid Id { get; set; }
    [Required]
    [StringLength(30)]
    public string Brand { get; set; }
    [Required]
    [StringLength(50)]
    public string Model { get; set; }
    public int DoorsNumber { get; set; }
    public int LuggageCapacity { get; set; }
    public int EngineCapacity { get; set; }
    public FuelType FuelType { get; set; }
    [DataType(DataType.Date)]
    public DateTime ProductionDate { get; set; }
    public double CarFuelConsumption { get; set; }
    public BodyType BodyType { get; set; }
}